src/app/models/player.py

```diff
@@ -32,29 +32,32 @@ async def get_kc(self, player_names: tuple[str]):
             # Create aliases for the tables
             reporting_player: dbPlayer = aliased(dbPlayer, name="reporting_player")
             reported_player: dbPlayer = aliased(dbPlayer, name="reported_player")
+            report_db: dbReport = aliased(dbReport, name="report_db")
 
             query: Select = select(
                 [
                     func.count(func.distinct(reported_player.id)).label("count"),
                     reported_player.possible_ban,
                     reported_player.confirmed_ban,
                     reported_player.confirmed_player,
-                    dbReport.manual_detect,
+                    report_db.manual_detect,
                 ]
             )
-            query = query.select_from(dbReport)
+            query = query.select_from(report_db)
             query = query.join(
-                reporting_player, dbReport.reportingID == reporting_player.id
+                reporting_player, report_db.reportingID == reporting_player.id
             )
             query = query.join(
-                reported_player, dbReport.reportedID == reported_player.id
+                reported_player, report_db.reportedID == reported_player.id
             )
-            query = query.where(reporting_player.name.in_(player_names))
+            query = query.where(
+                text("reporting_player.name IN :names").bindparams(names=player_names)
+            )  # prevent sql injection
             query = query.group_by(
                 reported_player.possible_ban,
                 reported_player.confirmed_ban,
                 reported_player.confirmed_player,
-                dbReport.manual_detect,
+                report_db.manual_detect,
             )
             result: Result = await self.session.execute(query)
             await self.session.commit()

```

Comment:
Please revert the changes in this file

Reply:
I changed it to use the alias as it improves performance greatly is there another name structure you would like to use?
https://github.com/Bot-detector/public-api/blob/4c1180aa78f47297df01280f7c21c719b9b79d35/src/app/models/player.py#L35-L45

Reply:
Interesting, i don't immediatly understand how it would speed up to query speed.
what is the generated sql before and after the change?

Reply:
yea. they are the exact same in cost.  i may have been conflating the performance with using aliases with the performance i gained with feedback score subquery command.

```mysql
SELECT count(distinct(reported_player.id)) AS count, reported_player.possible_ban, reported_player.confirmed_ban, reported_player.confirmed_player, report_db.manual_detect 
FROM `Players` AS player_db, `Reports` AS report_db INNER JOIN `Players` AS reporting_player ON report_db.`reportingID` = reporting_player.id INNER JOIN `Players` AS reported_player ON report_db.`reportedID` = reported_player.id 
WHERE player_db.name IN ('player1', 'player2', 'player3') GROUP BY reported_player.possible_ban, reported_player.confirmed_ban, reported_player.confirmed_player, report_db.manual_detect

-- 50ms through 82ms

SELECT count(distinct(reported_player.id)) AS count, reported_player.possible_ban, reported_player.confirmed_ban, reported_player.confirmed_player, `Reports`.manual_detect 
FROM `Players` AS player_db, `Reports` INNER JOIN `Players` AS reporting_player ON `Reports`.`reportingID` = reporting_player.id INNER JOIN `Players` AS reported_player ON `Reports`.`reportedID` = reported_player.id 
WHERE player_db.name IN ('player1', 'player2', 'player3') GROUP BY reported_player.possible_ban, reported_player.confirmed_ban, reported_player.confirmed_player, `Reports`.manual_detect

-- 50ms through 82ms
```

Reply:
[cleanup aliases](https://github.com/Bot-detector/public-api/pull/25/commits/c29eb904c219cbc01906f1cc27c6174b3378cc50)